tutorials/scripting/gdextension/gdextension_c_example.rst

```diff
@@ -0,0 +1,2165 @@
+
+So let's implement those two functions in the ``gdexample.c`` file:
+
+.. code-block:: c
+
+    void *gdexample_get_virtual_with_data(void *p_class_userdata, GDExtensionConstStringNamePtr p_name)
+    {
+        // If it is the "_process" method, return a pointer to the gdexample_class_process function.
+        if (is_string_name_equal(p_name, "_process"))
+        {
+            return (void *)gdexample_class_process;
+        }
+        // Otherwise, return NULL.
+        return NULL;
+    }
+
+    void gdexample_call_virtual_with_data(GDExtensionClassInstancePtr p_instance, GDExtensionConstStringNamePtr p_name, void *p_virtual_call_userdata, const GDExtensionConstTypePtr *p_args, GDExtensionTypePtr r_ret)
+    {
+        // If it is the "_process" method, call it with a helper.
+        if (p_virtual_call_userdata == &gdexample_class_process)
+        {
+            ptrcall_1_float_arg_no_ret(p_virtual_call_userdata, p_instance, p_args, r_ret);
+        }
+    }
+
+Those functions are also quite straightforward after making all the helpers
+previously.
+
+For the first one, we simply check if the function name requested is
+``_process`` and if it is we return a function pointer to our implementation of
+it. Otherwise we return ``NULL``, signaling that the method is not being
+overridden. We don't use the ``p_class_userdata`` here since this function is
+meant only for one class and we don't have any data associated with it.
+
+The second one is similar. If it is the ``_process`` method, it uses the given
+function pointer to call the ``ptrcall`` helper, passing the call arguments
+forward. Otherwise it simply does nothing, since we don't have any other virtual
+methods being implemented.
+
+The only thing missing is using those callbacks when the class is registered. Go
+to the ``init.c`` file and change the ``class_info`` initialization to include
+those, replacing the ``NULL`` value used previously:
+
+.. code-block:: c
+
+    void initialize_gdexample_module(void *p_userdata, GDExtensionInitializationLevel p_level)
+    {
+        ...
+
+        GDExtensionClassCreationInfo2 class_info = {
+            ...
+            .get_virtual_call_data_func = gdexample_get_virtual_with_data,
+            .call_virtual_with_data_func = gdexample_call_virtual_with_data,
+            ...
+        };
+
+        ...
+    }
+
+This is enough to bind the virtual method. If you build the extension and run
+the demo project again, the ``_process`` function will be called. You just won't
+be able to tell since the function itself does nothing visible. We will solve
+this now by making the custom node move following a pattern.
+
+In order to make our node do stuff, we'll need to call Godot methods. Not only
+the GDExtension API functions as we've being doing so far, but actual engine
+methods, as we would do with scripting. This naturally requires some extra setup.
+
+First, let's add :ref:`Vector2 <class_Vector2>` to our ``defs.h`` file, so we
```

Comment:
```suggestion
First, let's add :ref:`class_Vector2` to our ``defs.h`` file, so we
```
The explicit link title *can* be omitted here but in many existing cases it is not. So this suggestion is optional.